source/brailleDisplayDrivers/brltty.py

```diff
@@ -21,18 +21,41 @@
 
 KEY_CHECK_INTERVAL = 50
 
+# BRLAPI named pipes for local connections are numbered and all start with a common name.
+# This name is set on compile time and should be the same for all BRLTTY releases,
+# however, if a user compiles their own version this may differ
+BRLAPI_NAMED_PIPE_PREFIX = "BrlAPI"
+
 class BrailleDisplayDriver(braille.BrailleDisplayDriver):
 	"""brltty braille display driver.
 	"""
 	name = "brltty"
 	description = "brltty"
+	isThreadSafe = True
+
+	# Type info for auto property: _get_brlapi_pipes
+	brlapi_pipes: List[str]
+
+	@classmethod
+	def _get_brlapi_pipes(cls) -> List[str]:
+		"""Get the BrlAPI named pipes
+
+		Every BRLTTY instance with the BrlAPI enabled will have it's own named pipe to accept API connections.
+		The brlapi.Connection constructor takes either a `host:port` argument or just `:port`.
+		If only a port is given, this corresponds to the number at the end of the named pipe.
+		"""
+		return [pipe.name for pipe in os.scandir("//./pipe/") if pipe.name.startswith(BRLAPI_NAMED_PIPE_PREFIX)]

```

Comment:
have you considered using glob.glob instead?

Reply:
Yes, but glob.glob returns the full path. I thought it would make more sense to only return the file name, since pipes are in a known location anyway. and I think a list comprehension is more readable and probably even more efficient then filtering paths using a glob pattern and stripping the `\\.\pipe\` part from them.

I'm open for suggestions what this function should return though. Now it is just cast to a boolean, but it might be useful in the future for port selection.

Reply:
Thanks for clarifying, sounds reasonable!